internal/services/securitycenter/advanced_threat_protection_resource.go

```diff
@@ -88,6 +88,36 @@ func resourceAdvancedThreatProtectionCreateUpdate(d *pluginsdk.ResourceData, met
 		return fmt.Errorf("updating Advanced Threat protection for %q: %+v", id.TargetResourceID, err)
 	}
 
+	deadline, _ := ctx.Deadline()
```

Comment:
we should always ensure the context has a deadline:

```suggestion
	deadline, ok := ctx.Deadline()
	if !ok {
		return fmt.Errorf("internal-error: context had no deadline")
	}
```